python/lsst/pipe/tasks/dcrAssembleCoadd.py

```diff
@@ -716,24 +694,19 @@ def calculateConvergence(self, dcrModels, bbox, tempExpRefList, imageScalerList,
                                                                    bufferSize=self.bufferSize)
         if np.max(significanceImage) == 0:
             significanceImage += 1.
-        tempExpName = self.getTempExpDatasetName(self.warpType)
         weight = 0
         metric = 0.
         metricList = {}
-        zipIterables = zip(tempExpRefList, weightList, imageScalerList, spanSetMaskList)
-        for tempExpRef, expWeight, imageScaler, altMaskSpans in zipIterables:
-            exposure = tempExpRef.get(tempExpName + "_sub", bbox=bbox)
-            imageScaler.scaleMaskedImage(exposure.maskedImage)
-            singleMetric = self.calculateSingleConvergence(dcrModels, exposure, significanceImage, statsCtrl,
-                                                           altMaskSpans=altMaskSpans)
-            metric += singleMetric*expWeight
+        for tempExpRef, expWeight in zip(tempExpRefList, weightList):
+            exposure = subExposures[tempExpRef.dataId["visit"]][bbox]
+            singleMetric = self.calculateSingleConvergence(dcrModels, exposure, significanceImage, statsCtrl)
+            metric += singleMetric
             metricList[tempExpRef.dataId["visit"]] = singleMetric
-            weight += expWeight
+            weight += 1.
```

Comment:
I'm confused: you're adding both `singleMetric` and `1` to `weight` here. Why both?

Reply:
`singleMetric` is being added to `metric`, not `weight`.